src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -254,15 +307,16 @@ public override bool Equals(object obj)
         public bool Equals(ReadOnlyMemory<T> other)
         {
             return
-                _arrayOrOwnedMemory == other._arrayOrOwnedMemory &&
+                _object == other._object &&
                 _index == other._index &&
                 _length == other._length;
         }
 
-        [EditorBrowsable( EditorBrowsableState.Never)]
+        /// <summary>Returns the hash code for this <see cref="ReadOnlyMemory{T}"/></summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public override int GetHashCode()
         {
-            return CombineHashCodes(_arrayOrOwnedMemory.GetHashCode(), (_index & RemoveOwnedFlagBitMask).GetHashCode(), _length.GetHashCode());
+            return CombineHashCodes(_object.GetHashCode(), (_index & RemoveOwnedFlagBitMask).GetHashCode(), _length.GetHashCode());
```

Comment:
The masking off the RemoveOwnedFlagBitMask bit is unnecessary.

Reply:
Removed